demos/python_demos/object_detection_demo_centernet/detector.py

```diff
@@ -0,0 +1,217 @@
+        model = IENetwork(model=model_path, weights=os.path.splitext(model_path)[0] + '.bin')
+
+        assert len(model.inputs) == 1, "Expected 1 input blob"
+        assert len(model.outputs) == 3, "Expected 3 output blobs"
+
+        self._input_layer_name = next(iter(model.inputs))
+
+        self._output_layer_names = sorted([item for item in iter(model.outputs)])
+
+        self._ie = ie
+        self._exec_model = self._ie.load_network(model, device)
+        self._threshold = threshold
+        self.infer_time = -1
+        _, _, self.input_height, self.input_width = model.inputs[self._input_layer_name].shape
+
+    @staticmethod
+    def get_affine_transform(center, scale, rot, output_size, inv=0):
+
+        def get_dir(src_point, rot_rad):
+            sn, cs = np.sin(rot_rad), np.cos(rot_rad)
+            src_result = [0, 0]
+            src_result[0] = src_point[0] * cs - src_point[1] * sn
+            src_result[1] = src_point[0] * sn + src_point[1] * cs
+            return src_result
+
+        def get_3rd_point(a, b):
+            direct = a - b
+            return b + np.array([-direct[1], direct[0]], dtype=np.float32)
+
+        if not isinstance(scale, np.ndarray) and not isinstance(scale, list):
+            scale = np.array([scale, scale], dtype=np.float32)
+
+        scale_tmp = scale
+        src_w = scale_tmp[0]
+        dst_w, dst_h = output_size
+
+        rot_rad = np.pi * rot / 180
+        src_dir = get_dir([0, src_w * -0.5], rot_rad)
+        dst_dir = np.array([0, dst_w * -0.5], dtype=np.float32)
+
+        dst = np.zeros((3, 2), dtype=np.float32)
+        src = np.zeros((3, 2), dtype=np.float32)
+        src[0, :], src[1, :] = center, center + src_dir
+        dst[0, :] = [dst_w * 0.5, dst_h * 0.5]
+        dst[1, :] = np.array([dst_w * 0.5, dst_h * 0.5], np.float32) + dst_dir
+        src[2:, :] = get_3rd_point(src[0, :], src[1, :])
+        dst[2:, :] = get_3rd_point(dst[0, :], dst[1, :])
+
+        if inv:
+            trans = cv2.getAffineTransform(np.float32(dst), np.float32(src))
+        else:
+            trans = cv2.getAffineTransform(np.float32(src), np.float32(dst))
+
+        return trans
+
+    @staticmethod
+    def _gather_feat(feat, ind):
+        dim = feat.shape[1]
+        ind = np.expand_dims(ind, axis=1)
+        ind = np.repeat(ind, dim, axis=1)
+        feat = feat[ind, np.arange(feat.shape[1])]
+        return feat
+
+    @staticmethod
+    def _tranpose_and_gather_feat(feat, ind):
+        feat = np.transpose(feat, (1, 2, 0))
+        feat = feat.reshape((-1, feat.shape[2]))
+        feat = Detector._gather_feat(feat, ind)
+        return feat
+
+    @staticmethod
+    def _topk(scores, K=40):
+        cat, height, width = scores.shape
+
+        scores = scores.reshape((cat, -1))
+        topk_inds = np.argpartition(scores, -K, axis=1)[:, -K:]
+        topk_scores = scores[np.arange(scores.shape[0])[:, None], topk_inds]
+
+        topk_inds = topk_inds % (height * width)
```

Comment:
```suggestion

```
`scores` has a shape of `[cat, (height * width)]`, so the `topk_inds`'s values already can't be greater than height * width